minicluster/src/main/java/tachyon/master/AbstractLocalTachyonCluster.java

```diff
@@ -108,14 +108,16 @@ private void waitForMasterReady() {
     long startTime = System.currentTimeMillis();
     String actionMessage = "waiting for master to serve web";
     LOG.info(actionMessage + ELLIPSIS);
+    // The port should be set properly after the server has started
     while (!NetworkAddressUtils.isServing(getMaster().getWebBindHost(),
-        getMaster().getWebLocalPort())) {
+        getMaster().getWebLocalPort()) || mMasterConf.getInt(Constants.MASTER_WEB_PORT) == 0) {
```

Comment:
`isServing` is currently purely a network address util - it shouldn't need to worry about the way we reset `TachyonConf`s, which is more an implementation detail of how we start up the test cluster